riscv/interactive.cc

```diff
@@ -448,6 +452,50 @@ void sim_t::interactive_pc(const std::string& cmd, const std::vector<std::string
       << zext(get_pc(args), max_xlen) << std::endl;
 }
 
+reg_t sim_t::get_insn(const std::vector<std::string>& args)
+{
+  if (args.size() != 1)
+    throw trap_interactive();
+
+  processor_t *p = get_core(args[0]);
+  reg_t addr = p->get_state()->pc;
+  mmu_t* mmu = p->get_mmu();
+  reg_t val;
+
+  switch (addr % 8)
+  {
+    case 0:
+      val = mmu->load<uint64_t>(addr);
+      break;
+    case 4:
+      val = mmu->load<uint32_t>(addr);
+      break;
+    case 2:
+    case 6:
+      val = mmu->load<uint16_t>(addr);
+      break;
+    default:
+      val = mmu->load<uint8_t>(addr);
+      break;
+  }
+  return val;
+}
```

Comment:
Currently this is a bit duplicated with `get_pc` and `get_mem`, but this is significantly faster since reusing those functions requires you to re-create a `vector<string>`'s to use them.

Reply:
I think it's OK... no more sinful than everything else in `interactive.cc`.

Reply:
Actually, can you do some light refactoring?  Hoist this out as its own function `static reg_t load(mmu_t* mmu, reg_t addr)` so you can share the code between the two places without perf loss.

Reply:
Done.